byway/src/main/webapp/routepage.js

```diff
@@ -49,27 +71,110 @@ function calcRoute(directionsService, directionsRenderer, start, end) {
   });
 }
 
-function addToStops(stop){
-  console.log("add" + stop.placename + "to the ArrayList in the servlet");
+/** Clear the stops panel in the html */
+function clearStops(){
+  const stopList = document.getElementById('stop-list');
+  if(stopList != null){
+    stopList.innerText = "";
+  }
 }
 
-function getRecs() {
-  fetch('/api/recs')
+/** Get the new list of stops from datastore onload */
+function getStopsOnload(){
+  clearStops();
+  fetch('/api/stop')
   .then(response => response.json())
-  .then((recs) => {
-    const recList = document.getElementById('rec-list');
-    recList.innerText = ""; // clear old comments 
-    recs.forEach((stop) => {
-      let btn = document.createElement("button");
-      btn.innerText = stop.placename;
-      btn.setAttribute("class", "btn rec-btn");
-      btn.addEventListener("click", function() {
-        addToStops(stop);
-      });
-      recList.appendChild(btn);
+  .then((stopsResponse) => {
+    stopsResponse.forEach((stop)=>{
+      stops.push(stop);
     });
+    renderStopsList();
+  });
+}
+
+/** Re-render stop list synchronously */
+function renderStopsList(){
+  clearStops();
+  stops.forEach((stop)=>{
+    renderStop(stop);
+  })
+}
+
+/** Render stop list  
+ *  @param {String} stop a String to add as a button in the schedule panel in the html
+ */
+function renderStop(stop){
+  const stopList = document.getElementById('stop-list');
+  let btn = document.createElement('button');
+  btn.innerText = stop;
+  btn.className =  "btn rec-btn";
+  btn.addEventListener("click", function() {
+    // TODO: delete from recs later for better visuals on html
+    stops = stops.filter(function(stopObj){
+    return stopObj != stop;
+    })
+    updateStops(stop);
+  });
+  stopList.appendChild(btn);
+}
+
+/** Add stop to or delete stop from the stoplist in javascript and in the datastore
+ *  @param {String} stop a String to add or delete
+ */
+function updateStops(){
+  renderStopsList();
+  let stopsAsJSONString = JSON.stringify(stops);
+  let params = new URLSearchParams();
+  params.append("stops", stopsAsJSONString);
+  fetch('/api/stop', {method: 'POST', body:params});
```

Comment:
When sending json data in the body, it should be sufficient to just put `body: JSON.stringify(stops)`. You can ask Leo about how we retrieve this a JSON request body, since that was covered in #18

Reply:
Okay, will remove the extra variables. I also changed the servlet code to resemble what Leo had, but I'll read through your exchange in more depth.